src/components/__tests__/TheWelcome.spec.ts

```diff
@@ -1,48 +1,77 @@
-import { describe, it, expect, vi } from 'vitest';
-import { flushPromises, mount } from '@vue/test-utils';
+import { describe, it, expect, vi, beforeAll, afterAll, afterEach } from 'vitest';
+import { cleanup, fireEvent, render, screen } from '@testing-library/vue';
 import TheWelcomeVue from '../TheWelcome.vue';
 import axios from 'axios';
+import { server } from '@/mocks/handlers';
 
-const mockUserList = [
-  {
-    id: 1,
-    username: 'Klavenj',
-    isAdmin: false,
-  },
-  {
-    id: 2,
-    username: 'BillyJean',
-    isAdmin: false,
-  },
-  {
-    id: 5,
-    username: 'Anthony',
-    isAdmin: false,
-  },
-  {
-    id: 7,
-    username: 'Jeff',
-    isAdmin: false,
-  },
-  {
-    id: 8,
-    username: 'Grant',
-    isAdmin: true,
-  },
-];
+// Start server before all tests
+beforeAll(() => server.listen({ onUnhandledRequest: 'error' }));
+
+//  Close server after all tests
+afterAll(() => server.close());
+
+// Reset handlers after each test `important for test isolation`
+afterEach(() => server.resetHandlers());
 
 describe('TheWelcome.vue', async () => {
-  it('should load users on button click', async () => {
-    vi.spyOn(axios, 'get').mockResolvedValue(mockUserList);
 
-    const wrapper = mount(TheWelcomeVue);
+  it('should render button', async () => {
+    render(TheWelcomeVue);
+    const button = await screen.getByTestId('user-btn');
+    expect(button).toBeDefined;
+
+    cleanup();
+  });
+
+  it('should call axios two times after the button is clicked twice', async () => {
+    vi.spyOn(axios, 'get');
+    render(TheWelcomeVue);
+
+    const button = await screen.getByTestId('user-btn');
+
+    await fireEvent.click(button);
+    await fireEvent.click(button);
+
+    expect(axios.get).toHaveBeenCalledTimes(2);
+
+    cleanup();
+  });
+
+  it('should call axios with the correct request url', async () => {
+    vi.spyOn(axios, 'get');
+    render(TheWelcomeVue);
+
+    const button = await screen.getByTestId('user-btn');
 
-    wrapper.get('[data-test="user-btn"]').trigger('click');
+    await fireEvent.click(button);
 
-    expect(axios.get).toHaveBeenCalledTimes(1);
     expect(axios.get).toHaveBeenCalledWith('http://localhost:3000/users');
 
-    // Wait until the DOM updates.
-    await flushPromises();
+    cleanup();
   });
+
+  it('should render a list of users with the length of 7 when the button is clicked', async () => {
+    render(TheWelcomeVue);
+
+    const button = await screen.getByTestId('user-btn');
+    fireEvent.click(button);
+
+    const userList = await screen.findAllByTestId('users');
+    expect(userList.length).toBe(7);
+
+    cleanup();
+  });
+
+  it('should render a list containing one user named Klay when the button is clicked', async () => {
+    render(TheWelcomeVue);
+
+    const button = await screen.getByTestId('user-btn');
+    await fireEvent.click(button);
+
+    const userList = await screen.findAllByText('Klay');
+    expect(userList.length).toBe(1);
+    cleanup();
+  });
```

Comment:
I have a bit a problem with these tests. first off, is the idea that you shouldn't mock what you don't own, which is a principle discussed in the book Growing Object Oriented Software. What if axios suddenly changed it's api. Now everywhere, you're using it in your tests needs to change. How might you get around this?

The second issue is that you now also have to rely on MSW to test your API since it's difficult to create a response on your mock of axios. This introduces uneeded complexity and confusion to your tests. When do you mock axios? when do you use MSW? Do you have your msw handler correctly set-up, or are you mocking axios when you shouldn't. Things can get messy.